src/Raven.Server/Documents/Includes/IncludeRevisionsCommand.cs

```diff
@@ -40,9 +40,7 @@ public void Fill(Document document)
             {
                 var doc = _database.DocumentsStorage.RevisionsStorage.GetRevisionBefore(context: _context, id: document.Id, max: _revisionsBeforeDateTime.Value); 
                 if (doc is null) return; 
-                RevisionsChangeVectorResults ??= new Dictionary<string, Document>(StringComparer.OrdinalIgnoreCase); 
```

Comment:
The `Dictionary` has a `OrdinalIgnoreCase` comparor.
how can you have duplicates?

Reply:
also would it make sense to reuse existing code? e.g. `AddRevisionByDateTimeBefore` method?

Reply:
You have duplicates because you are putting it also in the dictionary `IdByRevisionsByDateTimeResults`

Reply:
Used AddRevisionByDateTimeBefore instead :)